lib/build-shim.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+var _ = require('lodash');
+var slice = Array.prototype.slice;
+var chalk = require('chalk');
+var warn = chalk.black.bgYellow;
+var parser = require('acorn').parse;
+var fs = require('fs');
+var path = require('path');
+var parse = require('./parse');
+
+/**
+ * Build requirejs shim object from bower dependencies object.
+ */
+module.exports = function (dependencies, opts, exclude) {
+  var shim = {};
+  // Stolen from ./parse.js
+  // @TODO: maybe put commonly used functions in one file
+  function filterName() {
+    var oldName = arguments[0];
+    var newName = _.difference(oldName.split('.'), slice.call(arguments, 1));
```

Comment:
If the purpose of the following few lines is just to remove the ".js" file extention, this can be simplified using [path.basename](http://nodejs.org/api/path.html#path_path_basename_p_ext) to something like...
`var newName = path.basename(oldName, '.js')`